site/server/test/PersistenceModuleSpec.scala

```diff
@@ -0,0 +1,454 @@
+  ): ConnectionIO[Int] =
+    Update[(String, Boolean)]("INSERT INTO persistence (name, active) VALUES (?,?)").updateMany(values)
+
+  def fetchAll: ConnectionIO[List[(String, Boolean, Long)]] =
+    sql"SELECT name,active,id FROM persistence".query[(String, Boolean, Long)].list
+
+  def fetchItemById(
+    id: Long
+  ): ConnectionIO[PersistenceItem] =
+    sql"SELECT id,name,active FROM persistence WHERE id=$id".query[PersistenceItem].unique
+
+  def fetchItemByStatus(
+    active: Boolean
+  ): ConnectionIO[List[PersistenceItem]] =
+    sql"SELECT id,name,active FROM persistence WHERE active=$active".query[PersistenceItem].list
+
+  def fetchItemByStatuses(
+    active: Boolean
+  ): ConnectionIO[List[PersistenceItem]] = {
+    val inactive = !active
+    sql"SELECT id,name,active FROM persistence WHERE active=$active OR active=$inactive".query[PersistenceItem].list
+  }
+
+  val persistenceModule = new PersistenceModule
+
+  def genBoundedList[T](minSize: Int = 1, maxSize: Int = 100, gen: Gen[T]): Gen[List[T]] =
+    Gen.choose(minSize, maxSize) flatMap { size ⇒ Gen.listOfN(size, gen) }
+
+  implicit val dataWithId: Arbitrary[List[(Long, String, Boolean)]] =
+    Arbitrary(
+      genBoundedList(
+        minSize = 2,
+        maxSize = 10,
+        gen = Gen.resultOf((l: Long, s: String, b: Boolean) ⇒ (l, s, b))
+      )
+    )
+
+  implicit val dataStringBoolean: Arbitrary[List[(String, Boolean)]] =
+    Arbitrary(genBoundedList(
+      minSize = 2,
+      maxSize = 10,
+      gen = Gen.resultOf((s: String, b: Boolean) ⇒ (s, b))
+    ))
+
+  implicit val dataStringBooleanLong: Arbitrary[List[(String, Boolean, Long)]] =
+    Arbitrary(genBoundedList(
+      minSize = 2,
+      maxSize = 10,
+      gen = Gen.resultOf((s: String, b: Boolean, l: Long) ⇒ (s, b, l))
+    ))
+
+  implicit val stringList: Arbitrary[List[String]] =
+    Arbitrary(genBoundedList(minSize = 2, gen = Gen.resultOf((s: String) ⇒ s)))
+
+  implicitly[Arbitrary[PersistenceItem]]
+
+  val persistenceItemGen = for {
+    pItem ← Arbitrary.arbitrary[PersistenceItem]
+    identifier ← Gen.identifier
+  } yield pItem.copy(name = identifier)
+}
+
+class PersistenceModuleSpec
+    extends WordSpec
+    with GeneratorDrivenPropertyChecks
+    with Matchers
+    with BeforeAndAfterEach
+    with DatabaseContext {
+
+  override def beforeEach = {
+    for {
+      _ ← dropTable
+      _ ← createTable
+    } yield ()
+  }.transact(trx).run
+
+  "fetchList (SQL without parameters)" should {
+    "return an empty list if the table is empty" in {
+      forAll(persistenceItemGen) { user ⇒
+        val list = persistenceModule.fetchList[PersistenceItem](
+          sql = fetchAllSql
+        ).transact(trx).run
+
+        list shouldBe empty
+      }
+    }
+  }
+
+  "return a list of PersistenceItem ((String, Boolean) tuple list) if there are some elements in the table " +
+    "that meet the criteria" in {
+      forAll(dataStringBoolean.arbitrary) { (data: List[(String, Boolean)]) ⇒
```

Comment:
All of this property need not to use the generator. They can be invoked directly by their Arbitrary evidences.